analyzers/tests/SonarAnalyzer.UnitTest/Helpers/GeneratedCodeRecognizerTest.cs

```diff
@@ -28,38 +28,40 @@ public class GeneratedCodeRecognizerTest
         [DataTestMethod]
         [DataRow(null)]
         [DataRow("")]
-        public void IsGenerated_WithNullOrEmptyPathAndNullRoot_ReturnsFalse(string path) =>
-            Assert_IsConsideredGenerated_ReturnsFalse(path);
+        public void IsGenerated_WithNullOrEmptyPathAndNullRoot_ReturnsFalse(string path)
+        {
+            var syntaxTree = new Mock<SyntaxTree>(MockBehavior.Loose);
+            syntaxTree.Setup(x => x.FilePath).Returns(path);
+
+            new TestRecognizer().IsGenerated(syntaxTree.Object).Should().BeFalse();
+        }
 
         [DataTestMethod]
         [DataRow("C:\\SonarSource\\SomeFile_razor.g.cs")]
         [DataRow("C:\\SonarSource\\SomeFile_cshtml.g.cs")]
         [DataRow("C:\\SonarSource\\SomeFile_razor.ide.g.cs")]
         [DataRow("C:\\SonarSource\\SomeFile_cshtml.ide.g.cs")]
         [DataRow("C:\\SonarSource\\SomeFile_RAZOR.g.cS")]
-        public void IsGenerated_RazorGeneratedFiles_ReturnsFalse(string path) =>
-            Assert_IsConsideredGenerated_ReturnsFalse(path);
-
-        [DataTestMethod]
-        [DataRow("C:\\SonarSource\\SomeFile.g.cs")]
-        [DataRow("C:\\SonarSource\\SomeFile_razor.g.cs.randomEnding")]
-        [DataRow("C:\\SonarSource\\SomeFile_cshtml.g.cs.randomEnding")]
-        [DataRow("C:\\SonarSource\\SomeFile_razor.g.ß")]
-        public void IsGenerated_NonRazorGeneratedFiles_ReturnsTrue(string path)
+        public void IsRazorGeneratedFile_RazorGeneratedFiles_ReturnsTrue(string path)
         {
+            // GetRoot() cannot be mocked - not virtual, so we use Loose behaviour to return null Root
             var syntaxTree = new Mock<SyntaxTree>(MockBehavior.Loose);
             syntaxTree.Setup(x => x.FilePath).Returns(path);
 
-            new TestRecognizer().IsConsideredGenerated(syntaxTree.Object).Should().BeTrue();
+            TestRecognizer.IsRazorGeneratedFile(syntaxTree.Object).Should().BeTrue();
```

Comment:
I would follow the IDE advice to use GeneratedCodeRecognizer instead of TestRecognizer